ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/OperandStack.java

```diff
@@ -24,71 +26,128 @@
  * <p>The operand stack is responsible for storing the current operands that the EVM can execute. It
  * is assumed to have a fixed size.
  */
-public interface OperandStack {
+public class OperandStack {
 
-  /**
-   * Returns the operand located at the offset from the top of the stack.
-   *
-   * @param offset the position relative to the top of the stack of the operand to return
-   * @return the operand located at the specified offset
-   * @throws IndexOutOfBoundsException if the offset is out of range (offset &lt; 0 || offset &gt;=
-   *     {@link #size()})
-   */
-  Bytes32 get(int offset);
+  private final Bytes32[] entries;
 
-  /**
-   * Removes the operand at the top of the stack.
-   *
-   * @return the operand removed from the top of the stack
-   * @throws IllegalStateException if the stack is empty (e.g. a stack underflow occurs)
-   */
-  Bytes32 pop();
+  private final int maxSize;
+
+  private int top;
+
+  public static class UnderflowException extends RuntimeException {
```

Comment:
Perhaps in a follow on PR.  If we pull the exceptions out I also want them to be checked since we are removing the callstack (for performance).  That way we either have to handle it or re-wrap it as a runnable.

Reply:
Sure, if we don't do it in this PR, we should make an issue so we can revisit it at some point. Something like `Unify Underflow and Overflow Signalling Across Stacks`.